Update Local Repo Search

pranath-reddy opened the pull request:
### Summary :memo:
This pull request introduces a minor update to the `LocalRepoCodeSearchTool`, focusing on improving search context, removing irrelevant repositories, and refreshing the vector database.

### Details
- Added repository descriptions alongside README text for richer search context.
- Removed non-astro/spam repositories using Ash’s curated astro repository list.
- Revectorized the dump and updated the dataset link.

**Note** – While GTE models performed best for planetary SME evaluation, results appeared inconsistent during manual testing, so the embedding model remains `MiniLM-L6-v2` for now.

### Checks
- [x] Tested Changes
- [ ] Stakeholder Approval

pranath-reddy committed:
Update Local Repo Search

- Filter Repo based on curation from Ash
- Add repo description to the context
- Revectorize the dump and update the GDrive link

muthukumaranR commented on akd/tools/code_search.py:
```diff
@@ -248,16 +248,19 @@ class LocalRepoCodeSearchToolConfig(CodeSearchToolConfig):
     """
 
-    data_file: str = str(get_akd_root() / "docs" / "repositories_with_embeddings.csv")
+    data_file: str = str(
+        get_akd_root() / "docs" / "repositories_with_embeddings_v2.csv"
+    )
     google_drive_file_id: str = os.getenv(
         "CODE_SEARCH_FILE_ID",
-        "1nPaEWD9Wuf115aEmqJQusCvJlPc7AP7O",
+        "1H15aL-y6DsQl-bt08cU0FaZf5NuwUzmD",
     )
     embedder_type: Literal["sentence-transformers", "openai"] = "sentence-transformers"
     wait_time: int = 1
     embedding_model_name: str = os.getenv("CODE_SEARCH_MODEL", "all-MiniLM-L6-v2")
     remove_embedding_column: bool = True
```
Let's use the `CombinedCodeSearchToolConfig` instead of using `env`. and add the defaults there.

NISH1001 replied:
@muthukumaranR But the ID is related to local embedding search only I think.

pranath-reddy replied:
Yes, the ID is specific to local code search. The combined tool is specifically for aggregating and reranking the results. Converting the PR to draft for now. I need to update the dump again by filtering repos based on word count.

NISH1001 reviewed (commented):
<comments/>

pranath-reddy committed:
update dump and sde tool

- Updated dump by filtering based on word count
- Updated sde code search tool to conform to the updated api endpoint
